File: app/src/main/java/com/benoitquenaudon/tvfoot/red/app/domain/match/state/MatchService.java
package com.benoitquenaudon.tvfoot.red.app.domain.match.state;

import com.benoitquenaudon.tvfoot.red.api.TvfootService;
import com.benoitquenaudon.tvfoot.red.app.data.entity.Match;
import io.reactivex.Single;
import javax.inject.Inject;

public class MatchService {
  private final TvfootService tvfootService;

  @Inject MatchService(TvfootService tvfootService) {
    this.tvfootService = tvfootService;
  }

  public Single<Match> loadMatch(String matchId) {
    return tvfootService.getMatch(matchId);
  }
}

File: app/src/main/java/com/benoitquenaudon/tvfoot/red/app/domain/match/state/MatchService.kt
package com.benoitquenaudon.tvfoot.red.app.domain.match.state

import com.benoitquenaudon.tvfoot.red.api.TvfootService
import com.benoitquenaudon.tvfoot.red.app.data.entity.Match
import io.reactivex.Single
import javax.inject.Inject

open class MatchService @Inject constructor(private val tvfootService: TvfootService) {
  fun loadMatch(matchId: String): Single<Match> = tvfootService.getMatch(matchId)
}
